src/Tuura/Concept/FSM/Translation.hs

```diff
@@ -205,6 +210,11 @@ fsmarcxToFsmarc arc = FsmArc newSourceEnc (transx arc) newDestEnc
     where newSourceEnc = (encToInt . srcEncx) arc
           newDestEnc = (encToInt . destEncx) arc
 
+removeInvariants :: [FsmArcX a] -> [[Tristate]] -> [FsmArcX a]
+removeInvariants xs inv = concatMap (\x -> if (destEncx x `elem` inv || srcEncx x `elem` inv)
+                          then []
+                          else [x]) xs
+
```

Comment:
That would be better, yes. Sorry, I'm still getting the hang of useful Haskell functions.